src/java.net.http/share/classes/jdk/internal/net/http/Exchange.java

```diff
@@ -421,8 +421,23 @@ private CompletableFuture<Response> checkFor407(ExchangeImpl<T> ex, Throwable t,
     }
 
     private CompletableFuture<Response> startSendingBody(DelegatingExecutor executor) {
-        return exchImpl.sendBodyAsync()
-                        .thenCompose(exIm -> exIm.getResponseAsync(executor));
+        // don't wait until the body has been fully sent to
+        // parse and return the response headers
+        var sendBody = exchImpl.sendBodyAsync();
+        CompletableFuture<Response> cf = exchImpl.getResponseAsync(executor);
+        sendBody.exceptionally((t) -> {
```

Comment:
Shouldn't this be `sendBody = sendBody.exceptionally(...)`?

Reply:
No - because if sendBody is completed exceptionally we want the dependent action to be executed. So `sendBody` should be the head, not the tail.